File: python/aoc/domain/compute/instructions/compare.py
from domain.compute.instructions.instruction import Instruction


class Compare(Instruction):
    @property
    def CompareFunction(self):
        return self._CompareFunction

    @property
    def Length(self):
        return 4

    def __init__(self, opcode, compare_function):
        super(Compare, self).__init__(opcode)
        self._CompareFunction = compare_function

File: Solution/Aoc/Aoc.Domain/Compute/Instructions/Compare.cs
using System;

namespace Aoc.Domain.Compute.Instructions
{
    internal class Compare : Instruction
    {
        public Func<int, int, bool> CompareFunction;
        public override int Length => 4;

        internal Compare(int opcode, Func<int, int, bool> compareFunction) : base(opcode)
        {
            CompareFunction = compareFunction;
            ParameterModes.Add(ParameterMode.Immediate);
        }
    }
}
